src/test/shared/vscode/window.ts

```diff
@@ -45,7 +45,12 @@ export function createTestWindow(): Window & TestWindow {
                         }
 
                         const d = onDidShowMessageEmitter.event(shownMessage => {
-                            if (shownMessage.message.match(expected)) {
+                            // HACK: the "rest" sniffing in TestMessage.create() (src/test/shared/vscode/message.ts)
+                            //       is broken for withProgress(). Workaround here: check `title` field.
+                            const title: string | undefined = (shownMessage.message as any)?.title
+                            const matchesOther = !!(title && title.match(expected))
```

Comment:
Yucky workaround because I don't understand what's going on in `TestMessage.create()`. But, this at least allows me to add tests (see `src/test/shared/utilities/messages.test.ts`).

@JadenSimon

Reply:
The VSC API heavily overloads `show[Information | Warning | Error]Message` which is what the ugly logic is handling. `showProgress` is unfortunately a special case that cannot be proxied with the same logic.

I almost have a PR ready to handle this.

Reply:
https://github.com/aws/aws-toolkit-vscode/pull/3182 handles this case.